x-pack/filebeat/docs/inputs/input-gcs.asciidoc

```diff
@@ -276,6 +280,8 @@ filebeat.inputs:
     - regex: '/Security-Logs/'
 ----
 
+NOTE: The `file_selectors` op is performed within the agent locally which scales vertically, hence using this option will cause the agent to download all the files and then filter them. This can cause a bottleneck in processing if the number of files is very high. It is recommended to use this attribute only when the number of files is limited or ample resources are available.
```

Comment:
It means like an ambiguity in `scales vertically` here:
Does it mean it supports multiple `file_selectors` regexes?
Multiple agents can be configured with same `bucket` and `file_selectors` regexes and they can distribute workload, like in SQS? I know this is not the case, but here, vertical scaling could be better explained.

Reply:
Since currently the file_selectors only filter our the file patters, this filtering occurs on the agent, hence the vertical scaling.